homeassistant/config.py

```diff
@@ -453,3 +455,31 @@ def merge_packages_customize(core_customize, packages):
         conf = schema(pkg)
         cust.extend(conf.get(CONF_CORE, {}).get(CONF_CUSTOMIZE, []))
     return cust
+
+
+@asyncio.coroutine
+def async_check_ha_config_file(hass):
+    """Check if HA config file valid.
+
+    This method is a coroutine.
+    """
+    import homeassistant.components.persistent_notification as pn
+
+    proc = yield from asyncio.create_subprocess_exec(
+        sys.argv[0],
+        '--script',
+        'check_config',
+        stdout=asyncio.subprocess.PIPE)
+    # Wait for the subprocess exit
+    (stdout_data, dummy) = yield from proc.communicate()
+    result = yield from proc.wait()
+    if result:
+        _LOGGER.error("check_config failed. Not restarting.")
```

Comment:
The "not restarting" part might no longer be relevant as there is a separate service to just check the config.